extensions/amp-ima-video/0.1/amp-ima-video.js

```diff
@@ -178,40 +178,43 @@ class AmpImaVideo extends AMP.BaseElement {
     const consentPromise = consentPolicyId
       ? getConsentPolicyState(element, consentPolicyId)
       : Promise.resolve(null);
-    return consentPromise.then((initialConsentState) => {
-      const iframe = getIframe(
-        win,
-        element,
-        'ima-video',
-        {initialConsentState},
-        {allowFullscreen: true}
-      );
-      iframe.title = this.element.title || 'IMA video';
-
-      this.applyFillContent(iframe);
-
-      // This is temporary until M74 launches.
-      // TODO(aghassemi, #21247)
-      addUnsafeAllowAutoplay(iframe);
-
-      this.iframe_ = iframe;
-
-      const deferred = new Deferred();
-      this.playerReadyPromise_ = deferred.promise;
-      this.playerReadyResolver_ = deferred.resolve;
-
-      this.unlistenMessage_ = listen(this.win, 'message', (e) =>
-        this.handlePlayerMessage_(/** @type {!Event} */ (e))
-      );
-
-      element.appendChild(iframe);
-
-      installVideoManagerForDoc(element);
-      Services.videoManagerForDoc(element).register(this);
-      observeContentSize(this.element, this.onResized_);
-
-      return this.loadPromise(iframe).then(() => this.playerReadyPromise_);
-    });
+    return consentPromise
+      .then((initialConsentState) =>
+        getIframe(
+          win,
+          element,
+          'ima-video',
+          {initialConsentState},
+          {allowFullscreen: true}
+        )
+      )
+      .then((iframe) => {
+        iframe.title = this.element.title || 'IMA video';
+
+        this.applyFillContent(iframe);
+
+        // This is temporary until M74 launches.
+        // TODO(aghassemi, #21247)
+        addUnsafeAllowAutoplay(iframe);
+
+        this.iframe_ = iframe;
+
+        const deferred = new Deferred();
+        this.playerReadyPromise_ = deferred.promise;
+        this.playerReadyResolver_ = deferred.resolve;
+
+        this.unlistenMessage_ = listen(this.win, 'message', (e) =>
+          this.handlePlayerMessage_(/** @type {!Event} */ (e))
+        );
+
+        element.appendChild(iframe);
+
+        installVideoManagerForDoc(element);
+        Services.videoManagerForDoc(element).register(this);
+        observeContentSize(this.element, this.onResized_);
+
+        return this.loadPromise(iframe).then(() => this.playerReadyPromise_);
+      });
```

Comment:
nit: don't nest promise chains

```suggestion
        return this.loadPromise(iframe)
      })
      .then(() => this.playerReadyPromise_);
```